src/wallet/wallet2.cpp

```diff
@@ -5989,6 +5993,33 @@ bool wallet2::is_output_blackballed(const crypto::public_key &output) const
   catch (const std::exception &e) { return false; }
 }
 
+bool wallet2::lock_keys_file()
+{
+  if (m_keys_file_locker)
+  {
+    MDEBUG(m_keys_file << " is already locked.");
```

Comment:
That should be an error, otherwise lock/lock/unlock/wait/unlock will actually unlock at the third step, and the wait will run while unlocked, which was not the intent.

Reply:
I'm not really sure. The purpose of this file locking is to prevent multiple wallet processes from opening the same wallet keys file simultaneously, and the wallet process will simply error out if the file is locked already; i.e. it won't wait for the file to be unlocked. The use case is quite different from mutex locking.

I think the current code doesn't include a code path that would call lock/unlock functions consecutively. I intended this null checking to be relevant mainly for the use by 3rd party via the API.